### PR TITLE
Add request time and description field in tethering request.

### PR DESCRIPTION
**Pull request by rmstar**
JIRA: CDAP-18765, CDAP-18717

### CODE REVIEW

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/PeerBase.java [hunk 1]
```diff
@@ -26,12 +26,16 @@
   private final String endpoint;
   private final TetheringStatus tetheringStatus;
   private final PeerMetadata metadata;
+  private final long requestTime;
 
-  public PeerBase(String name, @Nullable String endpoint, TetheringStatus tetheringStatus, PeerMetadata metadata) {
+
```
Comment (nliu132) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/PeerBase.java]: nit: extra new line

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/PeerMetadata.java [hunk 2]
```diff
@@ -27,10 +28,15 @@
   private final List<NamespaceAllocation> namespaceAllocations;
   // metadata associated with the peer. ex: project, region when peer is located
   private final Map<String, String> metadata;
+  // Text that describes this tethering
+  private final String description;
 
-  public PeerMetadata(List<NamespaceAllocation> namespaceAllocations, Map<String, String> metadata) {
+
```
Comment (nliu132) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/PeerMetadata.java]: nit: extra new line

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/PeerMetadata.java [hunk 3]
```diff
@@ -27,10 +28,15 @@
   private final List<NamespaceAllocation> namespaceAllocations;
   // metadata associated with the peer. ex: project, region when peer is located
   private final Map<String, String> metadata;
+  // Text that describes this tethering
+  private final String description;
```
Comment (nliu132) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/PeerMetadata.java:32]: do we have to support modifying the description?

Comment (rmstar) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/PeerMetadata.java:32]: No, atleast not in the MVP release. We also don't support editing metadata and namespace allocations.

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/PeerBase.java (hunk 1 above)
Comment (rmstar) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/PeerBase.java]: fixed

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/PeerMetadata.java (hunk 2 above)
Comment (rmstar) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/PeerMetadata.java]: fixed

**Review by nliu132 — APPROVED** [review]

**Pull request merged**